tests/trainer/optimization/test_manual_optimization.py

```diff
@@ -167,6 +167,7 @@ def training_epoch_end(self, outputs) -> None:
     with mock.patch.object(Strategy, "backward", wraps=trainer.strategy.backward) as bwd_mock:
         trainer.fit(model)
     assert bwd_mock.call_count == limit_train_batches * 3
+    assert trainer.global_step == limit_train_batches * 2
```

Comment:
the model has two optimizers and we call both of them, so this value makes sense. I'm not sure though which change in the loops is causing this / why it was 3 before? Is it because of fit_loop line 77?

Reply:
It wasn't 3 before, we didn't have tests for it but it would have been `== limit_train_batches`